ArraysAndStrings/isPalinPerm/solution2.js

```diff
@@ -0,0 +1,43 @@
+//function to check if there is a a palendrome in any permutation of string
+//ignore non-alpha char and spaces
+
+let yesP = 'qwertytrewq'
+let yesP2 = 'aabb'
+let emptyStr = ''
+let noP = 'qre'
+let space = 'tacca to'
+let ignoreNonLet = 'ta#$(3784cotac'
+let tooManyRepeats = 'aaac'
+let tooManyRepeatsTrue = 'aaccc'
+
+//time O(n), space O(n)
+function palendromePermutation(str) {
+  if (str.length === 0) return -1
+  let spaceRegex = /\s/g
+  let nonAlphaRegex = /[^a-z]/gi
+  str = str.replace(spaceRegex, '').replace(nonAlphaRegex, '').toLowerCase()
```

Comment:
From an interview prospective the use of regex is not preferred here.  A couple of reasons:
1. They are hard to remember/get right during an interview 
2. Your interviewer might not know regex off the top of their head and they will feel less confident in your answer if the code is not ran.

Since all you are doing is filtering out unwanted characters you can instead wrap your hash map add with an if that tests if each character is a valid character before adding it into the map.
Also small suggestion, I would try not resetting your function parameters in the function it's a bad pattern.

Reply:
Great, thanks for the feedback @yeomanse. I've updated my solution given your feedback. I'd love any other comments you have. @annamikmorozova It might be too close to your solution now? So I'd love further feedback on this so for my own repo I have a good solution, but happy to close this branch if it is too close.